tools/permissions.ts

```diff
@@ -0,0 +1,298 @@
+    target: PermissionTargetSchema.omit({ name: true })
+  })),
+  outputSchema: zodToJsonSchema(PermissionTargetSchema),
+  handler: async (args: any) => {
+    const { name, target } = args;
+    return await updatePermissionTarget(name, { name, ...target });
+  }
+};
+
+const deletePermissionTargetTool = {
+  name: "jfrog_delete_permission_target",
+  description: "Delete a permission target from the JFrog platform",
+  inputSchema: zodToJsonSchema(z.object({
+    name: z.string().describe("The name of the permission target to delete")
+  })),
+  outputSchema: zodToJsonSchema(z.object({
+    success: z.boolean()
+  })),
+  handler: async (args: any) => {
+    return await deletePermissionTarget(args.name);
+  }
+};
+
+// Resource type specific tools
+const getPermissionResourceTool = {
+  name: "jfrog_get_permission_resource",
+  description: "Get details of a specific resource type within a permission target",
+  inputSchema: zodToJsonSchema(z.object({
+    name: z.string().describe("The name of the permission target"),
+    resourceType: z.enum(['artifact', 'release_bundle', 'build']).describe("The type of resource to retrieve")
+  })),
+  outputSchema: zodToJsonSchema(z.union([ArtifactResourceSchema, ReleaseBundleResourceSchema, BuildResourceSchema])),
+  handler: async (args: any) => {
+    return await getPermissionResource(args.name, args.resourceType);
+  }
+};
+
+const updatePermissionResourceTool = {
+  name: "jfrog_update_permission_resource",
+  description: "Update a specific resource type within a permission target",
+  inputSchema: zodToJsonSchema(z.object({
+    name: z.string().describe("The name of the permission target"),
+    resourceType: z.enum(['artifact', 'release_bundle', 'build']).describe("The type of resource to update"),
+    resource: z.union([ArtifactResourceSchema, ReleaseBundleResourceSchema, BuildResourceSchema])
+  })),
+  outputSchema: zodToJsonSchema(z.union([ArtifactResourceSchema, ReleaseBundleResourceSchema, BuildResourceSchema])),
+  handler: async (args: any) => {
+    return await updatePermissionResource(args.name, args.resourceType, args.resource);
+  }
+};
+
+const replacePermissionResourceTool = {
+  name: "jfrog_replace_permission_resource",
+  description: "Replace a specific resource type within a permission target",
+  inputSchema: zodToJsonSchema(z.object({
+    name: z.string().describe("The name of the permission target"),
+    resourceType: z.enum(['artifact', 'release_bundle', 'build']).describe("The type of resource to replace"),
+    resource: z.union([ArtifactResourceSchema, ReleaseBundleResourceSchema, BuildResourceSchema])
+  })),
+  outputSchema: zodToJsonSchema(z.union([ArtifactResourceSchema, ReleaseBundleResourceSchema, BuildResourceSchema])),
+  handler: async (args: any) => {
+    return await replacePermissionResource(args.name, args.resourceType, args.resource);
+  }
+};
+
+const deletePermissionResourceTool = {
```

Comment:
Since delete operations can be risky, I suggest we hold off on including them for now. Let's comment this out until we gain more confidence in the tools and their behavior